pkg/cli/aws/aws.go

```diff
@@ -174,16 +185,19 @@ func refreshCredsViaOktaAWSCLI(ctx context.Context, copts *CredentialOptions, ac
 
 	args := []string{
 		"--open-browser",
-		"--write-aws-credentials",
 		"--cache-access-token",
 		"--profile",
 		copts.Profile,
-		"--aws-iam-role",
-		copts.Role,
+	}
+
+	if !copts.chooseRoleInteractively() {
+		args = append(args, "--aws-iam-role", copts.Role)
 	}
 
 	if acopts.Output == OutputCredentialProvider {
 		args = append(args, "--format", string(OutputCredentialProvider))
+	} else {
+		args = append(args, "--write-aws-credentials")
```

Comment:
This was a bug - `~/.aws/credentials` should not be written to when in credential provider mode.